x-pack/plugins/apm/public/components/app/trace_explorer/index.tsx

```diff
@@ -4,40 +4,40 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { EuiFlexGroup, EuiFlexItem, EuiTab, EuiTabs } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
 import React, { useEffect, useMemo, useState } from 'react';
 import { useHistory } from 'react-router-dom';
 import {
   TraceSearchQuery,
   TraceSearchType,
 } from '../../../../common/trace_explorer';
 import { useApmParams } from '../../../hooks/use_apm_params';
-import { useFetcher } from '../../../hooks/use_fetcher';
+import { useApmRouter } from '../../../hooks/use_apm_router';
+import { useApmRoutePath } from '../../../hooks/use_apm_route_path';
 import { useTimeRange } from '../../../hooks/use_time_range';
+import { TraceExplorerSamplesFetcherContextProvider } from '../../../hooks/use_trace_explorer_samples';
+import { APIClientRequestParamsOf } from '../../../services/rest/create_call_apm_api';
 import { ApmDatePicker } from '../../shared/date_picker/apm_date_picker';
-import { fromQuery, toQuery, push } from '../../shared/links/url_helpers';
-import { useWaterfallFetcher } from '../transaction_details/use_waterfall_fetcher';
-import { WaterfallWithSummary } from '../transaction_details/waterfall_with_summary';
+import { fromQuery, toQuery } from '../../shared/links/url_helpers';
+import { TechnicalPreviewBadge } from '../../shared/technical_preview_badge';
+import { TransactionTab } from '../transaction_details/waterfall_with_summary/transaction_tabs';
 import { TraceSearchBox } from './trace_search_box';
 
-export function TraceExplorer() {
+export function TraceExplorer({ children }: { children: React.ReactElement }) {
   const [query, setQuery] = useState<TraceSearchQuery>({
     query: '',
     type: TraceSearchType.kql,
   });
 
   const {
+    query: queryParams,
     query: {
       rangeFrom,
       rangeTo,
       environment,
       query: queryFromUrlParams,
       type: typeFromUrlParams,
```

Comment:
What do you think of?
```suggestion
  const {
    query: queryParams,
```
And then:
```
const {
      rangeFrom,
      rangeTo,
      environment,
      query: queryFromUrlParams,
      type: typeFromUrlParams,} = queryParams
```

Reply:
What's the benefit?

Reply:
It looks cleaner I'd say. And you wound't need to rename the query to `queryParams` you could basically use query.

Reply:
I already use `query` in the `useState` call at the top of the component though.